woocommerce/woocommerce-gateway-stripe
# Track whether the account was connected via the oauth flow

Fixes #3289 

## Changes proposed in this Pull Request:

This PR adds a flag that is stored in Stripe settings that records whether the test or live accounts were connected via an oauth flow. 

Using this flag we can communicate to the merchant if the account keys are set and connected or set but not connected.  

<p align="center">
<img width="599" alt="Screenshot 2024-07-19 at 12 54 37 PM" src="https://github.com/user-attachments/assets/636a3365-b07e-4130-aaa9-32f0779bcf83">
<img width="583" alt="Screenshot 2024-07-22 at 1 39 24 PM" src="https://github.com/user-attachments/assets/675877c4-1f80-4de5-b3d9-ab7d5b3e2a5d">
<br>
<sup>Account connected via OAuth (<strong>Top</strong>) or keys are set manually - incomplete (<strong>Bottom</strong>)</sup>
</p> 

## Testing instructions

1. Checkout this branch. 
2. Run `npm run build`
3. Go to your Stripe settings at **WooCommerce > Settings > Payments > Stripe (settings)**.
4. Open the account keys modal.
5. Note: 
   - if your keys have been set you will see **"Incomplete"**.
   - if your keys aren't set you will see **"Disconnected"**.
6. Go to the **Live** tab. 
7. Click the **Connect an account** button.
8. You will be redirected to the OAuth page hosted by Stripe. 
9. Login and select an account.
10. Click **Connect**.
11. At this point you can abandon the rest of the set up and just click the "Return to WooCommerce Inc." on the left.
12. Once you're back on the site, reopen the account keys modal.
13. You should see the account status is now **"Connected"**.

<p align="center">
<img width="634" alt="Screenshot 2024-07-19 at 1 06 16 PM" src="https://github.com/user-attachments/assets/956d655e-e1ab-4549-8174-0279baf01536">
</p> 

---


**Post merge**
